src/tools/elektrad/meta_handler.go

```diff
@@ -96,16 +102,105 @@ func (s *server) postMetaHandler(w http.ResponseWriter, r *http.Request) {
 	noContent(w)
 }
 
+// postMetaBulkHandler sets a whole set of metadata. In case there is a metakey with empty value it deletes the existing
+// Meta value.
+//
+// Arguments:
+//
+//	keyName the name of the key. URL path param.
+//	metaSet the set of metakeys for the given keyName
+//
+// Response Code:
+//
+//	201 No Content if the request is successful.
+//	401 Bad Request if no key name was passed - or the key name is invalid.
+//
+// Example: `curl -X POST -d '{ meta: [{"key": "hello", "value": "world"}] }' localhost:33333/kdbMetaBulk/user/test/hello`
+func (s *server) postMetaBulkHandler(w http.ResponseWriter, r *http.Request) {
+	var metaKeySet metaKeySet
+
+	keyName := parseKeyNameFromURL(r)
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&metaKeySet); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	if metaKeySet.MetaSet == nil {
+		badRequest(w)
+		return
+	}
+
+	errKey, err := elektra.NewKey(keyName)
+
+	if err != nil {
+		internalServerError(w)
+		return
+	}
+
+	defer errKey.Close()
+
+	parentKey, err := elektra.NewKey(keyName)
+
+	if err != nil {
+		badRequest(w)
+		return
+	}
+
+	defer parentKey.Close()
+
+	handle, ks := getHandle(r)
+
+	_, err = handle.Get(ks, errKey)
+
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+
+	k := ks.LookupByName(keyName)
+
+	if k == nil {
+		k = parentKey
+		ks.AppendKey(parentKey)
+	}
+
+	for _, meta := range metaKeySet.Meta {
+		if meta.Value == nil {
+			err = k.RemoveMeta(meta.Key)
+		} else {
+			err = k.SetMeta(meta.Key, *meta.Value)
+		}
+
+		if err != nil {
+        	writeError(w, err)
+        	return
+        }
+
+        err = set(handle, ks, errKey)
+
+        if err != nil {
+        	writeError(w, err)
+        	return
+        }
+	}
+
+	noContent(w)
+}
```

Comment:
```
func (s *server) postMetaBulkHandler(w http.ResponseWriter, r *http.Request) {
    metaKeySet, keyName, err := parseAndDecodeRequest(r)
    if err != nil {
        writeError(w, err)
        return
    }

    errKey, parentKey, err := createKeys(keyName)
    if err != nil {
        writeError(w, err)
        return
    }
    defer errKey.Close()
    defer parentKey.Close()

    handle, ks := getHandle(r)

    if err := getKeyFromHandle(handle, ks, errKey); err != nil {
        writeError(w, err)
        return
    }

    k := getKeyFromKeySet(ks, keyName, parentKey)

    if err := updateMetaKeys(handle, ks, errKey, k, metaKeySet.Meta); err != nil {
        writeError(w, err)
        return
    }

    noContent(w)
}
```

I didn't work a lot with `Go`, but I feel like creating some smaller functions could improve readability. Other than that the PR LGTM